packages/in_app_purchase/in_app_purchase_storekit/darwin/Classes/InAppPurchasePlugin.swift

```diff
@@ -0,0 +1,425 @@
+    #if os(iOS)
+      let messenger = registrar.messenger()
+    #endif
+    #if os(macOS)
+      let messenger = registrar.messenger
+    #endif
+    let channel = FlutterMethodChannel(
+      name: "plugins.flutter.io/in_app_purchase",
+      binaryMessenger: messenger)
+    let instance = InAppPurchasePlugin(registrar: registrar)
+    registrar.addMethodCallDelegate(instance, channel: channel)
+    registrar.addApplicationDelegate(instance)
+    SetUpInAppPurchaseAPI(messenger, instance)
+  }
+
+  @objc
+  public init(receiptManager: FIAPReceiptManager) {
+    self.receiptManager = receiptManager
+    self.requestHandlers = Set<FIAPRequestHandler>()
+    self.productsCache = NSMutableDictionary()
+    self.handlerFactory = { request in
+      return FIAPRequestHandler(request: request)
+    }
+    super.init()
+  }
+
+  @objc
+  public convenience init(
+    receiptManager: FIAPReceiptManager,
+    handlerFactory: @escaping (SKRequest) -> FIAPRequestHandler
+  ) {
+    self.init(receiptManager: receiptManager)
+    self.handlerFactory = handlerFactory
+  }
+
+  public convenience init(registrar: FlutterPluginRegistrar) {
+    self.init(receiptManager: FIAPReceiptManager())
+    self.registrar = registrar
+
+    self.paymentQueueHandler = FIAPaymentQueueHandler(
+      queue: SKPaymentQueue.default(),
+      transactionsUpdated: { [weak self] (transactions: [SKPaymentTransaction]) -> Void in
+        self?.handleTransactionsUpdated(transactions)
+      },
+      transactionRemoved: { [weak self] (transactions: [SKPaymentTransaction]) -> Void in
+        self?.handleTransactionsRemoved(transactions)
+      },
+      restoreTransactionFailed: { [weak self] error in
+        self?.handleTransactionRestoreFailed(error as NSError)
+      },
+      restoreCompletedTransactionsFinished: { [weak self] () -> Void in
+        self?.restoreCompletedTransactionsFinished()
+      },
+      shouldAddStorePayment: { [weak self] (payment: SKPayment, product: SKProduct) -> Bool in
+        return self?.shouldAddStorePayment(payment: payment, product: product) ?? false
+      },
+      updatedDownloads: { [weak self] _ in
+        self?.updatedDownloads()
+      }, transactionCache: FIATransactionCache())
+    #if os(iOS)
+      let messenger = registrar.messenger()
+    #endif
+    #if os(macOS)
+      let messenger = registrar.messenger
+    #endif
+    transactionObserverCallbackChannel = FlutterMethodChannel(
+      name: "plugins.flutter.io/in_app_purchase", binaryMessenger: messenger)
+  }
+
+  public func canMakePaymentsWithError(_ error: AutoreleasingUnsafeMutablePointer<FlutterError?>)
+    -> NSNumber?
+  {
+    return SKPaymentQueue.canMakePayments() as NSNumber
+  }
+
+  public func transactionsWithError(_ error: AutoreleasingUnsafeMutablePointer<FlutterError?>)
+    -> [SKPaymentTransactionMessage]?
+  {
+    let transactions = self.paymentQueueHandler!.getUnfinishedTransactions()
```

Comment:
is `paymentQueueHandler` guaranteed to be present?

Reply:
Technically yes? The other two initializers that dont set paymentQueueHandler are either marked unavailable or available in tests only in the objc version. I just looked it up and it doesnt look like theres a nice swift version for this

Reply:
nevermind, i saw your suggestion further below